packages/wrangler/src/pages.tsx

```diff
@@ -466,12 +467,20 @@ const generateAssetsFetch = async (
         if ((asset = getAsset(`${cwd}/404.html`))) {
           deconstructedResponse.status = 404;
           deconstructedResponse.body = serveAsset(asset);
+          deconstructedResponse.headers.set(
+            "Content-Type",
+            getType(asset) || "application/octet-stream"
```

Comment:
in a future PR, it may be useful to log a warning here, because it's rare that "application/octet-stream" will be the desired header. (This will be extremely rare because mime is pretty comprehensive, but still)

Reply:
Yeah, maybe. It is the fallback that we serve on Pages production, but I can see why people might want to flag when we're hitting that.